roles/manage-ssh-private-keys/tasks/process-ssh-key.yml

```diff
@@ -0,0 +1,20 @@
+---
+
+- name: Create destination path {{ ssh_key.dest | dirname }}
+  file:
+    path: "{{ ssh_key.dest | dirname }}"
+    state: directory
+    owner: "{{ ssh_key.owner }}"
+    group: "{{ ssh_key.group }}"
+    mode: 0700
+  when:
+    - ssh_key.create_dest_dir is defined
+    - ssh_key.create_dest_dir
+
+- name: Copy SSH key {{ ssh_key.src | basename }}
+  copy:
+    src: "{{ ssh_key.src }}"
+    dest: "{{ ssh_key.dest }}"
+    owner: "{{ ssh_key.owner }}"
+    group: "{{ ssh_key.group }}"
+    mode: "{{ ssh_key.mode | default(0400) | int }}"
```

Comment:
The `mode` is listed as a string in the `file` module docs, so probably just remove the `| int` filter on this one.

Reply:
especially since they can also be specified as symbolic m ode - i.e.: `u+rwx`, etc.